app/models/calculation_result.rb

```diff
@@ -280,32 +281,46 @@ def capital_rows(prefix:)
     capital_row_items(prefix:).transform_values { |value| monetise(value) }
   end
 
-  def main_home_rows(prefix:)
-    main_home = capital_items(:properties, prefix).fetch(:main_home)
-    disregard = main_home.fetch(:net_equity) - main_home.fetch(:assessed_equity)
-    data = {
-      value: monetise(main_home.fetch(:value)),
-      mortgage: monetise(-main_home.fetch(:outstanding_mortgage)),
-      disregards: monetise(-disregard),
-    }
-
-    transaction_allowance = main_home.fetch(:transaction_allowance)
-    data[:deductions] = monetise(-transaction_allowance) if transaction_allowance.positive?
-
-    data
+  def additional_property_data(prefix:)
+    additional_property = capital_items(:properties, prefix)[:additional_properties].first
+    property_data(additional_property, property_type: :additional)
   end
 
-  def additional_property_rows(prefix:)
-    home = capital_items(:properties, prefix)[:additional_properties].first
+  def property_data(property, property_type:)
     data = {
-      value: monetise(home.fetch(:value)),
-      mortgage: monetise(-home.fetch(:outstanding_mortgage)),
+      value: property.fetch(:value),
+      mortgage: -property.fetch(:outstanding_mortgage),
+      transaction_allowance: -property.fetch(:transaction_allowance),
+      smod_allowance: -property.fetch(:smod_allowance),
+      main_home_disregard: -property.fetch(:main_home_equity_disregard),
     }
 
-    transaction_allowance = home.fetch(:transaction_allowance)
-    data[:deductions] = monetise(-transaction_allowance) if transaction_allowance.positive?
+    data.each_key do |key|
```

Comment:
would there ever be any value in creating a new data object with the relevant fields rather than modifying the existing one with `delete` ?

Reply:
I buy that in general, immutable objects are more predictable, so in many circumstances yes, that would make sense. But in this case we've only just created `data` a couple of lines ago, and we've created it for the sole purpose of making a return value for this method, so I don't feel bad about monkeying with it.